src/mercury_engine_data_structures/formats/bcskla.py

```diff
@@ -0,0 +1,134 @@
+
+import construct
+from construct.core import (
+    Array,
+    Check,
+    Computed,
+    Const,
+    Construct,
+    Container,
+    FocusedSeq,
+    If,
+    IfThenElse,
+    Int8ul,
+    Int16ul,
+    Int32ul,
+    Pointer,
+    Rebuild,
+    Struct,
+    Tell,
+    this,
+)
+
+from mercury_engine_data_structures.common_types import Float
+from mercury_engine_data_structures.construct_extensions.alignment import AlignTo
+from mercury_engine_data_structures.formats.base_resource import BaseResource
+from mercury_engine_data_structures.formats.property_enum import PropertyEnumDoubleUnsafe
+from mercury_engine_data_structures.game_check import Game
+
+
+def _rebuild_flags(ctx):
+    flags = 0
+    for i, val in enumerate(ctx.data):
+        if isinstance(val, Container):
+            flags += 2 ** i
+    return flags
+
+def _update_next_kfv(ctx):
+    ctx._._._._next_kfv_offset = ctx._new_eof
+
+KeyFramedValues_Dread = Struct(
+    timing_type=Int16ul,
+    count=Rebuild(Int16ul, construct.len_(this.timings)),
+    timings=IfThenElse(
```

Comment:
What about using PrefixexArray here?

Reply:
I think it has to be a separate field because it's also used as the length of `values`

Reply:
use an adapter to treat it as a single array

Reply:
by that do you mean turn this whole struct into the subcon of an adapter? or just the count + arrays?